src/main/scala/org/marvin/executor/actions/OnlineAction.scala

```diff
@@ -35,8 +35,9 @@ import scala.util.{Failure, Success}
 
 object OnlineAction {
   case class OnlineExecute(message: String, params: String)
-  case class OnlineReload(protocol:String)
-  case class OnlineReloadNoSave(protocol: String)
+  trait ReloadType
+  case class OnlineReload(protocol:String) extends ReloadType
+  case class OnlineReloadNoSave(protocol: String) extends ReloadType
```

Comment:
Half yes, it was a design decision. I decided to let it in OnlineAction to treat the reload no save logic in the case statement instead of an IF ELSE inside the OnlineReload message. Is there any downside of doing this way? The problem here is the protocol, it was let by mistake. I will fix. 

Btw, the ReloadNoSave message was removed from PredictorFSM.

Reply:
I think the problem here is about the meaning/reason that you created this message type. In this case I beliave that a simple if/else statement is enough clear about what you wanted once the "OnlineReloadNoSave" message doesn't show the expected behavior... (Load the last local saved artifacts)

Therefore, I suggest you to remove this message and instead of this use a OnlineReload message without a protocol and put the logic inside the OnlineAction (in the OnlineReload message case statement).

Reply:
If we're going in such level of detail I believe this refactory should be bigger. The behavior of RemoteReload is kind of mysterious. It is not explicit clear that it will just reload the last model saved locally when the protocol is "" (not null, but empty). The protocol parameter is overloaded with different responsibilities. It is the parameter responsible to hold the protocol id that needs to be loaded and also the parameter that indicates wether the last model saved locally should be loaded. How about if we create a new parameter or even a new method in the grpc stub? By doing that I believe we can end up with a more cohesive code.